core/lib/spree/testing_support/factories/shipment_factory.rb

```diff
@@ -8,11 +8,12 @@
 
     transient do
       shipping_method nil
+      select_shipping_rate false
     end
 
-    after(:create) do |shipment, evalulator|
-      shipping_method = evalulator.shipping_method || create(:shipping_method, cost: evalulator.cost)
-      shipment.add_shipping_method(shipping_method, true)
+    after(:create) do |shipment, evaluator|
+      shipping_method = evaluator.shipping_method || create(:shipping_method, cost: evaluator.cost)
+      shipment.add_shipping_method(shipping_method, evaluator.select_shipping_rate)
```

Comment:
I'm a bit nervous about these changes to the factories, but they were the simplest way to get tests passing (except for one I'll dig into next)